projects/plugins/jetpack/class.jetpack.php

```diff
@@ -4952,6 +4998,25 @@ public static function authorize_ending_authorized( $data ) {
 		self::handle_post_authorization_actions( $activate_sso, $do_redirect_on_error );
 	}
 
+	/**
+	 * Fires on the jetpack_site_registered hook and acitvates default modules
+	 */
+	public static function site_registered() {
+		$active_modules = Jetpack_Options::get_option( 'active_modules' );
+		if ( $active_modules ) {
+			self::delete_active_modules();
+
+			// If there was previously activated modules (a reconnection), re-activate them all including those that require a user, and do not re-activate those that have been deactivated.
+			self::activate_default_modules( 999, 1, $active_modules, null, null );
+		} else {
+			// On a fresh new connection, at this point we activate only modules that do not require a user connection.
+			self::activate_default_modules( false, false, array(), null, null, null, false );
```

Comment:
this is just wild to me :) usually a sign that things would be better served by specific methods that do specific things.

Reply:
I agree. The truth is this whole thing needs to be refactored, but I was dodging that because I didn't want to delay the user-less project even more. 

My feeling is that instead of doing this slightly better I should refactor it all, but I'd like to it later so we don't block the whole project

Reply:
But I'll look into it to see if this can be simplified a little bit

Reply:
@dereksmart I went back and forth trying a simpler solution but I ended up with pretty much the same approach.

The only difference would be not to add the additional parameters to the `get_modules...` methods and manually filter them with `array_filter` and calling `get_module()` inside the callback to check the module properties... it's pretty verbose and confusing as well...

So I think that if the current approach is not good enough, I should spend some time and refactor it all.